### PR TITLE
Sprite preview & export, scale all sprites, Mono compatibility, menu bar

### PR DESCRIPTION
**Pull request by nastys**

### CODE REVIEW

File: MikuMikuModel/Nodes/Sprites/SpriteSetNode.cs [hunk 1]
```diff
@@ -63,6 +63,40 @@ protected override void Initialize()
...
+                                sprite.X = sprite.X * factor;
+                                sprite.Y = sprite.Y * factor;
+                            }
+
+                            IsDirty = true;
+                        }
+                        else
+                            MessageBox.Show("Invalid factor.", "Miku Miku Model", MessageBoxButtons.OK, MessageBoxIcon.Error);
+
+                        break;
```
Comment (blueskythlikesclouds) [inline MikuMikuModel/Nodes/Sprites/SpriteSetNode.cs]: Remove this break and add it after `IsDirty = true;` so the input dialog keeps appearing till the user inputs a number in correct format.

File: MikuMikuModel/Nodes/Sprites/SpriteNode.cs [hunk 2]
```diff
@@ -79,6 +85,39 @@ private void CalculateNdcValues()
...
+                        case "bmp":
+                            imageFormat = ImageFormat.Bmp;
+                            break;
+
+                        default:
+                            throw new ArgumentException("Image format could not be detected", nameof(filePath));
+                    }
+                }
+
+                SpriteCropper.Crop(Data, FindParent<SpriteSetNode>().Data).Save(filePath, imageFormat);
```
Comment (blueskythlikesclouds) [inline MikuMikuModel/Nodes/Sprites/SpriteNode.cs]: Wrap the returning Bitmap object in a using statement to dispose of it immediately.

File: MikuMikuModel/GUI/Controls/SpriteViewControl.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        private static SpriteViewControl sInstance;
+
+        private Bitmap mBitmap;
+
+        public static SpriteViewControl Instance => sInstance ?? (sInstance = new SpriteViewControl());
+
+        public void SetBitmap(Bitmap bitmap)
+        {
+            mBitmap = bitmap;
```
Comment (blueskythlikesclouds) [inline MikuMikuModel/GUI/Controls/SpriteViewControl.cs]: Dispose the previous bitmap if it exists.

File: MikuMikuModel/Nodes/Sprites/SpriteSetNode.cs [hunk 4]
```diff
@@ -63,6 +63,40 @@ protected override void Initialize()
...
+                while (true)
+                {
+                    using (var inputDialog = new InputDialog { WindowTitle = "Scale all sprites", Input = input })
+                    {
+                        if (inputDialog.ShowDialog() != DialogResult.OK)
+                            break;
+
+                        double dfactor;
+
+                        if (Double.TryParse(input, System.Globalization.NumberStyles.AllowDecimalPoint | System.Globalization.NumberStyles.AllowThousands, System.Globalization.CultureInfo.InvariantCulture, out dfactor))
```
Comment (blueskythlikesclouds) [inline MikuMikuModel/Nodes/Sprites/SpriteSetNode.cs]: Why don't you use float's TryParse method instead?
You should also declare the variable in the out statement like so: `out float factor`

**Comment by blueskythlikesclouds** [comment]
What's that weird symbol at the start of commit name also?

**Comment by nastys** [comment]
> 
> 
> What's that weird symbol at the start of commit message also?

¯\_(ツ)_/¯

**Pull request merged**